### PR TITLE
Test the default Content-Type headers of Request and Response objects

### PR DESCRIPTION
**Pull request by andreubotella**

### CODE REVIEW

File: fetch/api/request/request-init-contenttype.any.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  const blob = new Blob([], { type: "" });
+  const request = requestFromBody(blob);
+  assert_equals(request.headers.get("Content-Type"), null);
+}, "Default Content-Type for Request with Blob body (empty type)");
+
+test(() => {
+  const blob = new Blob([], { type: "a/b; c=d" });
+  const request = requestFromBody(blob);
+  assert_equals(request.headers.get("Content-Type"), "a/b; c=d");
+}, "Default Content-Type for Request with Blob body (set type)");
```
Comment (domenic) [inline fetch/api/request/request-init-contenttype.any.js:32]: Can you test a non-canonicalized MIME type, e.g. `Text/HTML;Charset="utf-8"`? I can't remember whether this should canonicalize (as `text/html;charset=utf-8`) or be left as-is.

Similarly testing an invalid-but-ASCII MIME type (e.g. `"foo"`) and a Unicode MIME type (e.g. `"😋/😁"`) would be worthwhile IMO.

File: fetch/api/request/request-init-contenttype.any.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+}, "Default Content-Type for Request with buffer source body");
+
+promise_test(async () => {
+  const formData = new FormData();
+  formData.append("a", "b");
+  const request = requestFromBody(formData);
+  const boundary = (await request.text()).split("\r\n")[0].slice(2);
+  assert_equals(
+    request.headers.get("Content-Type"),
+    `multipart/form-data; boundary=${boundary}`,
```
Comment (domenic) [inline fetch/api/request/request-init-contenttype.any.js:47]: Super-awesome that the spec mandates a non-canonical MIME type here (with the space) which is inconsistent with the below two tests 😑

Comment (andreubotella) [reply fetch/api/request/request-init-contenttype.any.js:47]: See fetch#423. I'm not sure what browsers did when Hixie [introduced the boundary parameter to the spec in 2011](https://github.com/whatwg/html/commit/1340a971bcba6bb4e90c79c3bc7ca42f19558f88), but now they're consistent.

File: fetch/api/request/request-init-contenttype.any.js (hunk 1 above)
Comment (andreubotella) [reply fetch/api/request/request-init-contenttype.any.js:32]: MIME types in `Blob`s are known to be broken, see whatwg/mimesniff#43 and in particular https://github.com/w3c/FileAPI/issues/43#issuecomment-840441099. The general consensus in that issue seems to be that `Blob`s should parse and serialize the MIME type, and some tests were added in #7764, but the spec hasn't been changed yet. So I'd rather not add new tests that contradict the ones from #7764.

Comment (domenic) [reply fetch/api/request/request-init-contenttype.any.js:32]: Fair enough.

**Review by domenic — APPROVED** [review]

**Pull request merged**